git-lfs: broken cross-compilation

### Nixpkgs version

- Stable (25.11)

### Describe the bug

Host: aarch64, Target: x86-64.

```bash
nix log /nix/store/ql9scidxnmgbkcgghi6dc0djbf1ijwb2-git-lfs-3.7.1-go-modules-x86_64-unknown-linux-gnu.drv
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/28bb1dkbcsm1p20l551h4f544mdb7n23-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
go: downloading github.com/git-lfs/gitobj/v2 v2.1.1
go: downloading github.com/rubyist/tracerx v0.0.0-20170927163412-787959303086
go: downloading github.com/spf13/cobra v1.7.0
go: downloading github.com/git-lfs/pktline v0.0.0-20210330133718-06e9096e2825
go: downloading golang.org/x/sync v0.12.0
go: downloading github.com/pkg/errors v0.0.0-20170505043639-c605e284fe17
go: downloading github.com/git-lfs/wildmatch/v2 v2.0.1
go: downloading github.com/golang/groupcache v0.0.0-20241129210726-2c02b8208cf8
go: downloading github.com/leonelquinteros/gotext v1.5.0
go: downloading github.com/mattn/go-isatty v0.0.4
go: downloading github.com/olekukonko/ts v0.0.0-20171002115256-78ecb04241c0
go: downloading github.com/dpotapov/go-spnego v0.0.0-20210315154721-298b63a54430
go: downloading github.com/ssgelm/cookiejarparser v1.0.1
go: downloading golang.org/x/net v0.38.0
go: downloading github.com/jmhodges/clock v1.2.0
go: downloading golang.org/x/sys v0.31.0
go: downloading github.com/git-lfs/go-netrc v0.0.0-20250218165306-ba0029b43d11
go: downloading golang.org/x/text v0.23.0
go: downloading github.com/spf13/pflag v1.0.5
go: downloading github.com/jcmturner/gokrb5/v8 v8.4.2
go: downloading github.com/jcmturner/gofork v1.0.0
go: downloading github.com/hashicorp/go-uuid v1.0.2
go: downloading github.com/jcmturner/goidentity/v6 v6.0.1
go: downloading github.com/jcmturner/dnsutils/v2 v2.0.0
go: downloading golang.org/x/crypto v0.36.0
go: downloading github.com/jcmturner/aescts/v2 v2.0.0
go: downloading github.com/jcmturner/rpc/v2 v2.0.3
# command-line-arguments
loadinternal: cannot find runtime/cgo
/nix/store/f3cw6lrjpv9b55baz4lhy1sr0s3y9y37-go-1.25.4/share/go/pkg/tool/linux_arm64/link: running x86_64-unknown-linux-gnu-gcc failed: exit status 1
/nix/store/ymqk8pzzh9b26m3s7smm8rjwlgmnvriz-x86_64-unknown-linux-gnu-gcc-wrapper-14.3.0/bin/x86_64-unknown-linux-gnu-gcc -s -Wl,--build-id=0x4957bf1c6861e97c9eb49e8de110f48a9ddd701c -o $WORK/b001/exe/mangen -Wl,--compress-debug-sections=zlib /tmp/nix-build-git-lfs-3.7.1-go-modules-x86_64-unknown-linux-gnu.drv-0/go-link-991615672/go.o -no-pie
/nix/store/1h635gigah7y5icwpmh0hc71jh4ygd3r-x86_64-unknown-linux-gnu-binutils-2.44/bin/x86_64-unknown-linux-gnu-ld: /tmp/nix-build-git-lfs-3.7.1-go-modules-x86_64-unknown-linux-gnu.drv-0/go-link-991615672/go.o: Relocations in generic ELF (EM: 183)
/nix/store/1h635gigah7y5icwpmh0hc71jh4ygd3r-x86_64-unknown-linux-gnu-binutils-2.44/bin/x86_64-unknown-linux-gnu-ld: /tmp/nix-build-git-lfs-3.7.1-go-modules-x86_64-unknown-linux-gnu.drv-0/go-link-991615672/go.o: Relocations in generic ELF (EM: 183)
/nix/store/1h635gigah7y5icwpmh0hc71jh4ygd3r-x86_64-unknown-linux-gnu-binutils-2.44/bin/x86_64-unknown-linux-gnu-ld: /tmp/nix-build-git-lfs-3.7.1-go-modules-x86_64-unknown-linux-gnu.drv-0/go-link-991615672/go.o: Relocations in generic ELF (EM: 183)
/nix/store/1h635gigah7y5icwpmh0hc71jh4ygd3r-x86_64-unknown-linux-gnu-binutils-2.44/bin/x86_64-unknown-linux-gnu-ld: /tmp/nix-build-git-lfs-3.7.1-go-modules-x86_64-unknown-linux-gnu.drv-0/go-link-991615672/go.o: Relocations in generic ELF (EM: 183)
/nix/store/1h635gigah7y5icwpmh0hc71jh4ygd3r-x86_64-unknown-linux-gnu-binutils-2.44/bin/x86_64-unknown-linux-gnu-ld: /tmp/nix-build-git-lfs-3.7.1-go-modules-x86_64-unknown-linux-gnu.drv-0/go-link-991615672/go.o: Relocations in generic ELF (EM: 183)
/nix/store/1h635gigah7y5icwpmh0hc71jh4ygd3r-x86_64-unknown-linux-gnu-binutils-2.44/bin/x86_64-unknown-linux-gnu-ld: /tmp/nix-build-git-lfs-3.7.1-go-modules-x86_64-unknown-linux-gnu.drv-0/go-link-991615672/go.o: Relocations in generic ELF (EM: 183)
/nix/store/1h635gigah7y5icwpmh0hc71jh4ygd3r-x86_64-unknown-linux-gnu-binutils-2.44/bin/x86_64-unknown-linux-gnu-ld: /tmp/nix-build-git-lfs-3.7.1-go-modules-x86_64-unknown-linux-gnu.drv-0/go-link-991615672/go.o: Relocations in generic ELF (EM: 183)
/nix/store/1h635gigah7y5icwpmh0hc71jh4ygd3r-x86_64-unknown-linux-gnu-binutils-2.44/bin/x86_64-unknown-linux-gnu-ld: /tmp/nix-build-git-lfs-3.7.1-go-modules-x86_64-unknown-linux-gnu.drv-0/go-link-991615672/go.o: Relocations in generic ELF (EM: 183)
/nix/store/1h635gigah7y5icwpmh0hc71jh4ygd3r-x86_64-unknown-linux-gnu-binutils-2.44/bin/x86_64-unknown-linux-gnu-ld: /tmp/nix-build-git-lfs-3.7.1-go-modules-x86_64-unknown-linux-gnu.drv-0/go-link-991615672/go.o: Relocations in generic ELF (EM: 183)
/nix/store/1h635gigah7y5icwpmh0hc71jh4ygd3r-x86_64-unknown-linux-gnu-binutils-2.44/bin/x86_64-unknown-linux-gnu-ld: /tmp/nix-build-git-lfs-3.7.1-go-modules-x86_64-unknown-linux-gnu.drv-0/go-link-991615672/go.o: Relocations in generic ELF (EM: 183)
/nix/store/1h635gigah7y5icwpmh0hc71jh4ygd3r-x86_64-unknown-linux-gnu-binutils-2.44/bin/x86_64-unknown-linux-gnu-ld: /tmp/nix-build-git-lfs-3.7.1-go-modules-x86_64-unknown-linux-gnu.drv-0/go-link-991615672/go.o: error adding symbols: file in wrong format
collect2: error: ld returned 1 exit status

commands/commands.go:29: running "go": exit status 1
```

### Steps to reproduce

```bash
nix build github:NixOS/nixpkgs/25.11#pkgsCross.gnu64.git-lfs
```

### Expected behaviour

Should work

### Screenshots

_No response_

### Relevant log output

```console

```

### Additional context

_No response_

### System metadata

```bash
nix-shell -p nix-info --run "nix-info -m"
warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels' does not exist, ignoring
 - system: `"aarch64-linux"`
 - host os: `Linux 6.16.0, NixOS, 25.05 (Warbler), 25.05.20250523.11cb351`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.28.3`
 - nixpkgs: `/nix/store/5j5y90a6m6jssvg2xybf86xgkjfa4zab-source`

```

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/gi/git-lfs/package.nix
+++ b/pkgs/by-name/gi/git-lfs/package.nix
@@ -21,7 +21,8 @@ buildGoModule (finalAttrs: {
     hash = "sha256-N5ckTnyA3mueZre+rMhFZBiAFgEu4pmtzkiUidXnan8=";
   };
 
-  vendorHash = "sha256-6H0KpLin+DqwEg5bdzaxj2CoNSneZ/ET43MTrrdF3h8=";
+  proxyVendor = true;
+  vendorHash = "sha256-SUnZ9uN43CAw/iHC8cPBm3nYD03d3Pg2pYS2PwjDCnE=";
 
   nativeBuildInputs = [
     asciidoctor
@@ -37,7 +38,7 @@ buildGoModule (finalAttrs: {
   subPackages = [ "." ];
 
   preBuild = ''
-    GOARCH= go generate ./commands
+    CC= GOOS= GOARCH= go generate ./commands
   '';
 
   postBuild = ''

